spec/bundler/source/git/git_proxy_spec.rb

```diff
@@ -136,8 +136,11 @@
         expect(subject).not_to receive(:git)
 
         expect { subject.copy_to(destination, submodules) }.
-          to raise_error(Bundler::Source::Git::MissingGitRevisionError,
-            "Revision #{revision} does not exist in the repository #{uri}. Maybe you misspelled it?")
+          to raise_error(
+            Bundler::Source::Git::MissingGitRevisionError,
+            "Git error: command `git command` in directory #{destination} has failed.\n" \
```

Comment:
:thinking: to make this assertion closer to the real scenario you can maybe change line 135 to (command can be stored to variable to be used twice):

```ruby
git_command = "reset --hard #{revision}"
expect(subject).to receive(:git).with(git_command).and_raise(Bundler::Source::Git::GitCommandError, git_command)
```

And assert `git reset --hard...` is part of the error message.

Reply:
Done in https://github.com/bundler/bundler/pull/7225/commits/ca96316b97353b4344ba283be0051ef7e9d9a013.